tests/run/t4866.scala

```diff
@@ -0,0 +1,27 @@
+// Test that try-finally aren't lifted, but try-catch are.
+
+class Foo(param: => Object) {
```

Comment:
Do we need a by-name parameter? Let's minimise feature interaction in tests:
```scala
class Foo {
 def foo: Int = try ??? finally ()
}
```

Reply:
We don't. Removed it.